core/src/main/java/org/infinispan/util/concurrent/locks/impl/StripedLockContainer.java

```diff
@@ -34,10 +34,11 @@ public StripedLockContainer(int concurrencyLevel) {
    }
 
    @Inject
-   void inject(@ComponentName(KnownComponentNames.BLOCKING_EXECUTOR) Executor executor, TimeService timeService) {
+   void inject(@ComponentName(KnownComponentNames.BLOCKING_EXECUTOR) Executor blockingExecutor,
+         @ComponentName(KnownComponentNames.NON_BLOCKING_EXECUTOR) Executor nonBlockingExecutor, TimeService timeService) {
       for (int i = 0; i < sharedLocks.length; i++) {
          if (sharedLocks[i] == null) {
-            sharedLocks[i] = new InfinispanLock(executor, timeService);
+            sharedLocks[i] = new InfinispanLock(blockingExecutor, nonBlockingExecutor, timeService);
```

Comment:
We should set the executors on the other branch as well. Even though I'd rather avoid that other branch completely, there's too much code in the tests that relies on replacing components after startup.

Reply:
I think I will just remove the if/else block.

Reply:
I'm not sure that's ok either, as it will basically drop the locks that are held by in-progress operations/txs, and the dropped locks will never resume their waiters.

And unfortunately modules that need to override core components use the same `rewire()` functionality as tests, so we can't say that it will only affect tests (although modules should do this before any locks can be acquired, there's no guarantee).